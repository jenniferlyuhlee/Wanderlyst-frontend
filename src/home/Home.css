.Home-title{
    padding: 2rem;
    margin: 2rem;
    width: 60%;
    background-color: rgba(0, 0, 0, 0.4);
    border-radius: 20px;
}

@media (max-width: 1000px){
    .Home-title{
        width: 90%;
    }
    .Home{
        flex-direction: column;
    }
}

.Home-header{
    display: flex;
    align-items: center;
}

.Home-gif{
    width: 150px;
}

.btn{
    border-radius: 25px;
}

