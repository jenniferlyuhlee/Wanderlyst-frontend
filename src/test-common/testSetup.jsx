import React, { useState } from "react";
import UserContext from "../auth-user/UserContext";
import WanderlystApi from "../utils/api";
import '@testing-library/jest-dom';

const demoUser = {
  username: "testadmin", 
  firstName: "testfirst",
  lastName: "testlast",
  location: "testloc",
  bio: "testbio",
  profilePic: null,
  isAdmin: true,
};

const demoTags = [
  {id: 1, name: 'Foodie', description: 'Savor the world one bite at a time. Dive into a journey packed with delicious local cuisines, hidden culinary gems, and must-try dishes that will leave your taste buds tingling.'},
  {id: 2, name: 'Explorer', description: 'Step off the beaten path and uncover the unknown. From ancient ruins to vibrant cityscapes, let your curiosity guide you through unforgettable experiences.'}
]

const UserProvider = ({ children, 
                        currUser = demoUser, 
                        setCurrUser, 
                        tags = demoTags,
                        logout,
                        hasLikedItin,
                        toggleLike }) => {
  const [user, setUser] = useState(currUser);

  return (
    <UserContext.Provider value={{ currUser: user, 
                                  setCurrUser: setUser, 
                                  tags, logout, 
                                  hasLikedItin, toggleLike}}>
      {children}
    </UserContext.Provider>
  );
};

export { UserProvider };