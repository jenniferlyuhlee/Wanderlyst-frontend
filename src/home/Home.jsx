import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth-user/UserContext";
import pickRandom from "../helpers/random";
import "./Home.css"

/** Home page component
 *  Displays different content based on currUser
 *  Includes functionality to navigate to random tag detail page
 */

function Home(){
    const { currUser, tags } = useContext(UserContext);

    return(
        <div className = "Home">
            <div className = "Home-title">
                {currUser ?
                <>
                    <div className = "Home-header">
                        <h1 className = "display-1 wanderlyst fw-medium">
                            Welcome {currUser.firstName}!
                        </h1>
                        <img className = "Home-gif" 
                            src = "https://www.futurewings.co.za/wp-content/uploads/2023/07/giphy.gif" />
                    </div>
                    <p className = "fs-6 mb-2">
                        Your go-to guide for planning and sharing itineraries that inspire.
                        Connect with journeys that fuel your desire to wander every corner of the world.
                    </p>
                    <Link to = "/explore" 
                        className = "rounded-pill btn btn-lg btn-primary me-2">
                        Search itineraries
                    </Link>
                    <p className = "fs-6 mt-4 mb-2">
                        Don't know where to start? Click below to explore a random tag.
                    </p>
                    <Link to = {`/tags/${pickRandom(tags.map(t => t.name))}`}
                        className = "rounded-pill btn btn-lg btn-primary">
                        Explore a tag
                    </Link>
                </>
                :
                <>
                    <h1 className = "display-1 wanderlyst fw-medium">
                        WanderLyst
                    </h1>
                    <p className = "fs-6 mb-2">
                        Your go-to guide for planning and sharing itineraries that inspire.
                        Connect with journeys that fuel your desire to wander every corner of the world.
                    </p>
                    <Link to = "/signup" 
                        className = "rounded-pill btn btn-lg btn-primary">
                        Join now
                    </Link>
                </>
                }
            </div>
        </div>
    )
}

export default Home;