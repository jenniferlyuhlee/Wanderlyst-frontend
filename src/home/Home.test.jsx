import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { UserProvider } from "../test-common/testSetup";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import Home from "./Home";
import pickRandom from "../helpers/random";
import { beforeEach, afterEach, expect, describe, test, vi } from "vitest";

describe("Home component: currUser", () => {
    
    // mock random func in link randomizer for consistent snapshot tests
    vi.mock("../helpers/random", () => ({
        __esModule: true,
        default: vi.fn(() => "Foodie"),
    }));

    // restore mocks
    afterEach(() => {
        vi.restoreAllMocks();
    });

    test("renders as without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <Home />
                </UserProvider>
            </MemoryRouter>)
    });
    test("matches snapshot", () => {
        const {asFragment} = render(
                            <MemoryRouter>
                                <UserProvider>
                                    <Home/>
                                </UserProvider>
                            </MemoryRouter>)
        expect(asFragment()).toMatchSnapshot();
    });
    test("renders home for logged in users", () => {
        const {getByText} = render(
                            <MemoryRouter>
                                <UserProvider currUser>
                                    <Home/>
                                </UserProvider>
                            </MemoryRouter>)
        expect(getByText("Explore a tag")).toBeInTheDocument()
    });
});


describe("Home component: anon", () => {
    test("renders as without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider currUser = {null}>
                    <Home />
                </UserProvider>
            </MemoryRouter>)
    });
    test("matches snapshot", () => {
        const {asFragment} = render(
                            <MemoryRouter>
                                <UserProvider currUser = {null}>
                                    <Home/>
                                </UserProvider>
                            </MemoryRouter>)
        expect(asFragment()).toMatchSnapshot();
    });
    test("renders home for anonymous users", () => {
        const {getByText} = render(
                            <MemoryRouter>
                                <UserProvider currUser = {null}>
                                    <Home/>
                                </UserProvider>
                            </MemoryRouter>)
        expect(getByText("Join now")).toBeInTheDocument()
    });
});