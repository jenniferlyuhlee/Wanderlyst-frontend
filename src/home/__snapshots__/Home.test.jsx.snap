// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component: anon > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="Home-title"
    >
      <h1
        class="display-1 wanderlyst fw-medium"
      >
        WanderLyst
      </h1>
      <h4>
        <b>
          wan·der·lyst (/ˈwändərˌlist/)
        </b>
        <br />
        a strong desire to travel and share one's 
        <i>
          lysts
        </i>
      </h4>
      <p
        class="fs-6 mb-2"
      >
        Your go-to guide for planning and sharing itineraries that inspire. Connect with journeys that fuel your desire to wander every corner of the world.
      </p>
      <a
        class="rounded-pill btn btn-lg btn-primary"
        href="/signup"
      >
        Join now
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Home component: currUser > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <div
      class="Home-title"
    >
      <div
        class="Home-header"
      >
        <h1
          class="display-1 wanderlyst fw-medium"
        >
          Welcome testfirst!
        </h1>
        <img
          class="Home-gif"
          src="https://www.futurewings.co.za/wp-content/uploads/2023/07/giphy.gif"
        />
      </div>
      <p
        class="fs-6 mb-2"
      >
        Your go-to guide for planning and sharing itineraries that inspire. Connect with journeys that fuel your desire to wander every corner of the world.
      </p>
      <a
        class="rounded-pill btn btn-lg btn-primary me-2"
        href="/explore"
      >
        Search itineraries
      </a>
      <p
        class="fs-6 mt-4 mb-2"
      >
        Don't know where to start? Click below to explore a random tag.
      </p>
      <a
        class="rounded-pill btn btn-lg btn-primary"
        href="/tags/Foodie"
      >
        Explore a tag
      </a>
    </div>
  </div>
</DocumentFragment>
`;
